src/engine/GroupBy.h

```diff
@@ -257,20 +258,21 @@ class GroupBy : public Operation {
         for (const auto& aggregate : alias.aggregateInfo_) {
           ++numAggregates;
 
-          auto addIf = [this]<typename T, HashMapAggregateType target>(
+          auto addIf = [this]<typename T>(
+                           std::type_identity<T>, HashMapAggregateType target,
                            const HashMapAggregateInformation& info) {
             if (info.aggregateType_.type_ == target)
               aggregationData_.emplace_back(
                   sparqlExpression::VectorWithMemoryLimit<T>{alloc_});
           };
 
-          addIf.template operator()<AvgAggregationData, AVG>(aggregate);
-          addIf.template operator()<CountAggregationData, COUNT>(aggregate);
-          addIf.template operator()<MinAggregationData, MIN>(aggregate);
-          addIf.template operator()<MaxAggregationData, MAX>(aggregate);
-          addIf.template operator()<SumAggregationData, SUM>(aggregate);
-          addIf.template operator()<GroupConcatAggregationData, GROUP_CONCAT>(
-              aggregate);
+          using namespace ad_utility::use_type_identity;
+          addIf(TI<AvgAggregationData>, AVG, aggregate);
+          addIf(TI<CountAggregationData>, COUNT, aggregate);
+          addIf(TI<MinAggregationData>, MIN, aggregate);
+          addIf(TI<MaxAggregationData>, MAX, aggregate);
+          addIf(TI<SumAggregationData>, SUM, aggregate);
+          addIf(TI<GroupConcatAggregationData>, GROUP_CONCAT, aggregate);
```

Comment:
Two very minor suggestions which make this clearer and safer:
1.  You can capture `aggregate` inside the `addIf` lambda, then you get rid of the redundant last argument (purely cosmetic).
2. Add a check that exactly one of your addIfs was called (easiest: `AD_CORRECTNESS_CHECK` that the size of the `aggregationData_` increases by exactly one during that chain.